src/drt/src/dr/FlexDR_maze.cpp

```diff
@@ -2787,6 +2800,33 @@ void FlexDRWorker::setNDRStyle(drNet* net,
   }
 }
 
+void FlexDRWorker::editStyleExt(frSegStyle& currStyle,
+                                frMIdx startX,
+                                frMIdx endX,
+                                frMIdx startY,
+                                frMIdx endY,
+                                frMIdx z,
+                                FlexMazeIdx* prev,
+                                FlexMazeIdx* next)
+{
+  auto layer = getTech()->getLayer(gridGraph_.getLayerNum(z));
+  if (layer->getWrongDirWidth() >= layer->getWidth())
+    return;
+  bool vertical = startX == endX;
+  bool nextVertical = next && endX == next->x();
+  bool prevVertical = prev && startX == prev->x();
+  if (layer->isHorizontal() == vertical)
+    return;
+  if (next && next->z() == z && vertical != nextVertical
```

Comment:
I'd suggest turning these ifs into helper functions. They have identical structure so you could have a small function that you could call twice that takes a FlexMazeIdx* and a frMIdx and call it twice in this function